webview-ui/src/components/mcp/configuration/tabs/add-server/AddRemoteServerForm.tsx

```diff
@@ -65,16 +35,26 @@ const AddRemoteServerForm = ({ onServerAdded }: { onServerAdded: () => void }) =
 		}
 
 		setError("")
-
 		submittedValues.current = { name: serverName.trim() }
-
 		setIsSubmitting(true)
 		setShowConnectingMessage(true)
-		vscode.postMessage({
-			type: "addRemoteServer",
-			serverName: serverName.trim(),
-			serverUrl: serverUrl.trim(),
-		})
+
+		try {
+			await McpServiceClient.addRemoteMcpServer({
+				serverName: serverName.trim(),
+				serverUrl: serverUrl.trim(),
+			})
+			setIsSubmitting(false)
```

Comment:
```suggestion
                         setMcpServers(mcpServers)
			setIsSubmitting(false)
```

We need to set the servers frontend state

Reply:
Good call, added it here: https://github.com/cline/cline/pull/3147/files#diff-331617007ed7380f97f590b32bd87b2bf2a18e1c74e24790a8a5e18b2d28a04fR50